galleries/examples/subplots_axes_and_figures/axhspan_demo.py

```diff
@@ -29,6 +29,7 @@
 ax2.axvspan(mu+sigma, mu+2*sigma, color='0.95')
 ax2.axvline(mu, color='darkgrey', linestyle='--')
 ax2.plot(x, y)
+ax2.set(title="axvspan")
```

Comment:
Side note, probably not enough to change in hindsight, but to set future practice:

I would generally favor `set_thing(x)` over `set(thing=x)` for single properties.

And in this particular example, I would also have kept the two properties `ax1.set(ylim=(-1.5, 1.5), title="axhspan")` above separate. Mainly because of symmetry with `ax2.set_title` and to a lesser degree because ylim and title are quite different properties.

Reply:
`Axes.set()` is similar to `pyplot`. It's a convenience interface that people can use to simplify their code - if they know what they are doing, but it does not provide the full capability. I'd say: If you only want to learn or teach the minimal stuff, stick with `set_*()`. Only when it's a real simplification, use `set()`. - Possibly this is something we should discuss and agree on a strategy, like wie did with Axes-interface vs. pyplot. It's not helpful if the docs do it one way or the other depending on the author.

Reply:
Thanks for digging that up :rocket:. I'm sure I advocated for "advertise `set()` more". But I don't agree with

> one-shot set aimed at use cases who are using us as an application and the set_XYZ for when we are being used internal to a third-party library.

That's not the criterion for me. Instead, I would like to advertise `set()` as a shortcut. As API lead I see the need and call for a discussion on how we consider and in which cases we want to advertise `set()`.

Reply:
The set_title method is far more featureful and should usually be preferred to set(title=) unless there is some reason to be concise.  Using nonstandard patterns trips up reading an example and I don't think there is a strong argument to use "set" as a standard.

Reply:
Discussion moved to #28693